lib/invoiced/payment_source.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+module Invoiced
+  class PaymentSource < Object
+    include Invoiced::Operations::List
+
+    OBJECT_NAME = 'payment_source'
+
+    def list(params = {})
+      response = @client.request(:get, endpoint, params)
+
+      output = []
+
+      response[:body].each do |obj|
+        if obj[:object] == 'card'
+          output.push(Util.convert_to_object(@client, Invoiced::Card, obj, self))
+        end
+        if obj[:object] == 'bank_account'
+          output.push(Util.convert_to_object(@client, Invoiced::BankAccount, obj, self))
+        end
+      end
+
+      # store the metadata from the list operation
+      metadata = Invoiced::List.new(response[:headers][:link], response[:headers][:x_total_count])
+
+      [output, metadata]
+    end
+  end
+
+  class BankAccount < Object
+    include Invoiced::Operations::Create
+    include Invoiced::Operations::Delete
+
+    OBJECT_NAME = 'bank_account'
+
+    def create(body = {}, opts = {})
+      body['method'] = 'bank_account'
+
+      # change endpoint just for this operation
+      @endpoint = '/payment_sources'
+      output = super
+      @endpoint = '/bank_accounts'
+
+      @endpoint = @endpoint + '/' + @id.to_s if @id
+
+      output
+    end
+  end
+
+  class Card < Object
+    include Invoiced::Operations::Create
+    include Invoiced::Operations::Delete
+
+    OBJECT_NAME = 'card'
+
+    def create(body = {}, opts = {})
+      body['method'] = 'card'
+
+      # change endpoint just for this operation
+      @endpoint = '/payment_sources'
```

Comment:
I'm not a huge fan of this in case an exception happens the endpoint is not changed back. While it might still be ok it does feel flimsy. I think you should just re-implement create here instead of decorating it.

Reply:
Can do

Reply:
In doing this I discovered the `ensure` keyword in Ruby and am able to fix this possible issue without reimplementing `create`.